LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+Inside the `fabric-tools` folder, we can now start the Docker images that comprise the Hyperledger Fabric network
+
+```
+./startFabric.sh
+```
+
+Then we can create an Hyperledger Fabric peer administrator identity for our networks:
+```
+./createPeerAdminCard.sh
+```
+You can then start the Composer Playground, which will run on port 8080 of the VM:
+
+
+Finally, start Composer Playground
+
+```
+composer-playground
+```
+
+# Writing and Deploying a Business Network
+
+## Overview of the Tuna Business Network
+As shown previously, we will implement a simple network to track the movement of Tuna fish.
+
+The network we will build maintains a single system where fishers, restaurant owners and regulators interact.
+
+![Participants](resources/img_02-01.png)
+
+Each participant is able to access and work upon information about Tuna fish.
+
+Importantly, the blockchain enables this to happen in a way that is immutable and distributed, while enabling a degree of transparency and oversight not easily implementable in a centralised database.
+
+## Steps
+
+In order to create and use the `tuna-network` business network, we will cover the following steps:
+
+1) Creating an empty network
+2) Defining Participants
+3) Defining Assets and Transactions
+4) Developing Transaction Logic
+5) Developing Queries
+6) Defining Access Control Rules
+7) Building and starting the Business Network
+8) Deploying onto Hyperledger Fabric
+9) Testing on the Composer Playground
+10) Running the Composer REST Server
+
+> The `tuna-network` Business Network can be downloaded at this repository: https://github.com/hyperledger/education/composer-material
+
+## 1) Creating an Empty Network
+
+We can use [Yeoman](http://yeoman.io/) to create an empty network, by running:
+
+```
+yo hyperledger-composer:businessnetwork
+```
+
+And then answering the questions that are posed.
+
+```
+? Business network name: tuna-network
+? Description: Hyperledger Composer network for Tuna tracking
+? Author name:  Alejandro (Sasha) Vicente Grabovetsky & Nicola Paoli
+? Author email: sasha@aid.technology, nicola@aid.technology
+? License: Apache-2.0
+? Namespace: org.tuna
+? Do you want to generate an empty template network? Yes: generate an empty template network
+```
+
+## 2) Defining Participants
+
+Participants are defined under the `models/org.tuna.cto` file.
+
+We start by defining a `namespace`.
+```
+namespace org.tuna
+```
+
+Then we create an abstract `Participant` for an `Individual`.
+All participants will inherit the properties from it.
+```
+abstract participant Individual identified by id {
+    o String id
+    o String name
+    o Address address
+}
+```
+
+To fill the property `Address` of the individual, we can create a `Concept`.<br>
+Note that `postCode` should have a specific format that can be validated using a regular expression.
+```
+concept Address {
+    o String addressLine
+    o String locality
+    o String postCode regex=/\d{4}[ ]?[A-Z]{2}/
```

Comment:
This is a strange postcode for people in the United States. Might want to call that out. When I created a Fisher in the playground, I got a message saying that my 85001 postcode did not conform to the regex. I am okay with not changing the expression, but it is probably worth pointing out that the expectation is 4 digits followed by 2 characters.

Reply:
Good point, fixed by adding text above explaining we are referring to Dutch postcode and explain what format it takes.